raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -914,6 +906,58 @@ def test_events_for_onchain_secretreveal():
     })
 
 
+def test_events_for_onchain_secretreveal_once():
+    """ Secret must be registered on-chain only once. """
+    amount = 10
+    block_number = 1
+    channel_map, transfers_pair = make_transfers_pair(
+        [HOP2_KEY, HOP3_KEY, HOP4_KEY, HOP5_KEY],
+        amount,
+        block_number,
+    )
+
+    pair = transfers_pair[0]
+    channel_identifier = pair.payer_transfer.balance_proof.channel_identifier
+    channel_state = channel_map[channel_identifier]
+
+    for channel_state in channel_map.values():
+        channel.register_offchain_secret(channel_state, UNIT_SECRET, UNIT_SECRETHASH)
+
+    block_number = (
+        pair.payer_transfer.lock.expiration - channel_state.reveal_timeout
+    )
+
+    events = mediator.events_for_onchain_secretreveal_if_dangerzone(
+        channel_map,
+        transfers_pair,
+        block_number,
+    )
+    assert len(events) == 1
```

Comment:
In this case I was testing for only one entry, because we don't want to send the secret reveal multiple times, I can filter the list for the event type and assert on the length of that if that is good?

Reply:
Understood. Since any code path in that function call will return 1 type of events then i guess checking for the count is good enough.